### PR TITLE
Add feature to change popup trigger event

### PR DESCRIPTION
**Pull request by anto-christo**
This PR is with reference to issue #59 

I've added a new function under Gantt class named `change_popup_mode`.
By default (i.e if you do not call the above function), the popup will be shown on 'click' event.

You can override it as follows:
`ganttObject.change_popup_mode(event);`  

(ganttObject is the object of Gantt class and event is the event which should trigger the popup.)

`change_popup_mode` can take any appropriate DOM event as it's parameter.

For example:
`ganttObject.change_popup_mode('mouseover');  //To show popup on hover`
`ganttObject.change_popup_mode('dblclick');  //To show popup on double-click`
`ganttObject.change_popup_mode('click');  //To show popup on click (default)`

### CODE REVIEW

File: src/svg_utils.js [hunk 1]
```diff
@@ -19,15 +19,15 @@ export function createSVG(tag, attrs) {
...
 
-export function animateSVG(svgElement, attr, from, to) {
+export function animateSVG(svgElement, attr, from, to, trigger_mode) {
     const animatedSvgElement = getAnimationElement(svgElement, attr, from, to);
 
     if (animatedSvgElement === svgElement) {
         // triggered 2nd time programmatically
         // trigger artificial click event
         const event = document.createEvent('HTMLEvents');
-        event.initEvent('click', true, true);
```
Comment (netchampfaris) [inline src/svg_utils.js]: you need not change `animateSVG` this has a different purpose.

File: src/index.js [hunk 2]
```diff
@@ -570,8 +577,8 @@ export default class Gantt {
         parent_element.scrollLeft = scroll_pos;
     }
 
-    bind_grid_click() {
-        $.on(this.$svg, 'click', '.grid-row, .grid-header', () => {
+    grid_event() {
```
Comment (netchampfaris) [inline src/index.js]: why change method name?

File: src/index.js [hunk 3]
```diff
@@ -139,6 +141,12 @@ export default class Gantt {
         this.change_view_mode();
     }
 
+    change_popup_mode(mode = this.options.popup_trigger){
```
Comment (netchampfaris) [inline src/index.js]: I don't think we need to provide a method to change the popup event on the fly. It should only be set on initialization

File: src/bar.js [hunk 4]
```diff
@@ -175,16 +175,14 @@ export default class Bar {
     }
 
     setup_click_event() {
-        $.on(this.group, 'focus click', e => {
+        $.on(this.group, this.gantt.options.popup_trigger, e => {
```
Comment (netchampfaris) [inline src/bar.js]: `focus` should be added always, for keyboard handling

**Comment by anto-christo** [comment]
@netchampfaris , I've made the requested changes to the best of my knowledge.

Now instead of creating a separate function, we'll have to pass popup_trigger in the 'options' parameter while initializing the object.

For example:
`var gantt_chart = new Gantt(".gantt-target", tasks, {
			popup_trigger: 'mouseover'
		});`

Can you please confirm whether this is what you meant by setting at the time of initialization?
And even in this case, the default trigger will be a 'click' event.

**Comment by jkramp** [comment]
This is a great addition but it appears to break the on_click handler (it no longer functions when using popup_trigger: 'mouseover'

**Pull request merged**